src/Domain/item/ItemDomain.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Domain.dao;
+using Domain.utils;
+using Dto;
+using Dto.tools;
+
+namespace Domain.item;
+
+public class ItemDomain : IItemDomain {
+
+    private readonly IItemDao _itemDao;
+
+    public ItemDomain(IItemDao itemDao) {
+        _itemDao = itemDao;
+    }
+
+    public async Task<ItemDto> CreateItem(ItemDto itemDto) {
+        itemDto = new ItemDto(itemDto.Name.Trim());
```

Comment:
I guess trim only remove white space , so we might also need to check with : lower case and uppercase

Reply:
The use case document says nothing about lowercase or uppercase.  But then again we can change it. But what exactly needs to be checked with lowercase or uppercase ? Is it we only allow item names on lowercase ?

Reply:
Now, the getItemByName checks for name ignoring the case. This should fix the issue we discussed about.